src/backend/app.js

```diff
@@ -21,10 +22,85 @@ app.use(cors());
 
 router.use("/meals", mealsRouter);
 
+//Respond with all meals in the future (relative to the when datetime)
+app.get("/future-meals", async (req, res) => {
+  try {
+    const today = new Date();
+    const futureMeals = await knex("Meal").where("_when", ">", today).select();
+    if (futureMeals.length === 0) {
+      res
+        .status(404)
+        .json({ data: null, message: "No future meal available." });
+      return;
+    }
+    res.status(200).json({ data: futureMeals, message: "ok" });
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ data: null, message: "server error" });
+  }
+});
+
+//past-meals 	Respond with all meals in the past (relative to the when datetime)
+app.get("/past-meals", async (req, res) => {
+  try {
+    const today = new Date();
+    const futureMeals = await knex("Meal").where("_when", "<", today).select();
+    if (futureMeals.length === 0) {
+      res
+        .status(404)
+        .json({ data: null, message: "No future meal available." });
+      return;
+    }
+    res.status(200).json({ data: futureMeals, message: "ok" });
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ data: null, message: "server error" });
+  }
+});
+//all-meals  Respond with all meals sorted by ID
+app.get("/all-meals", async (req, res) => {
+  try {
+    const allMeals = await knex("Meal").select().orderBy("id", "asc");
+
+    res.status(200).json({ data: allMeals, message: "ok" });
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ data: null, message: "server error" });
+  }
+});
+
+//first-meal 	Respond with the first meal (meaning with the minimum id)
+app.get("/first-meals", async (req, res) => {
+  try {
+    const firstMeal = await Meal.findOne({
+      order: [["id", "ASC"]],
+    });
+
+    res.status(200).json({ data: firstMeal, message: "ok" });
```

Comment:
The assigment requests that in cases where there should be just one meal returned, and there are no meals in the database, the return code should be `404`, and an appropriate message. ☝️